src/pages/Profile.js

```diff
@@ -154,6 +155,110 @@ const NihLink = ({ nihToken }) => {
 }
 
 
+const FenceLink = ({ provider, displayName }) => {
+  const decodeProvider = state => state ? JSON.parse(atob(state)).provider : ''
+
+  const extractToken = (provider, { state, code }) => {
+    const extractedProvider = decodeProvider(state)
+    return extractedProvider && provider === extractedProvider ? code : undefined
+  }
+
+  const queryParams = qs.parse(window.location.search, { ignoreQueryPrefix: true })
+  const token = extractToken(provider, queryParams)
+  const redirectUrl = `${window.location.origin}/${Nav.getLink('fence-callback')}`
+
+  /*
+   * Hooks
+   */
+  const [{ username, issued_at: issuedAt }, setStatus] = useState({})
+  const [href, setHref] = useState(undefined)
+  const [isLoadingStatus, setIsLoadingStatus] = useState(false)
+  const [isLoadingAuthUrl, setIsLoadingAuthUrl] = useState(false)
+  const [isLinking, setIsLinking] = useState(false)
+  const signal = useCancellation()
+
+  const { User } = Ajax(signal)
+
+  const loadAuthUrl = _.flow(
+    withErrorReporting(`Error loading Fence link`),
+    Utils.withBusyState(setIsLoadingAuthUrl)
+  )(async () => {
+    const result = await User.getFenceAuthUrl(provider, redirectUrl)
+    setHref(result.url)
+  })
+
+  const loadFenceStatus = _.flow(
+    withErrorReporting(`Error loading status for ${displayName}`),
+    Utils.withBusyState(setIsLoadingStatus)
+  )(async () => {
+    try {
+      setStatus(await User.getFenceStatus(provider))
+    } catch (error) {
+      if (error.status === 404) {
+        setStatus({})
+      } else {
+        throw error
+      }
+    }
+  })
+
+  const linkFenceAccount = _.flow(
+    withErrorReporting('Error linking NIH account'),
+    Utils.withBusyState(setIsLinking)
+  )(async () => {
+    setStatus(await User.linkFenceAccount(provider, token, redirectUrl))
+  })
+
+  Utils.useOnMount(() => {
+    loadAuthUrl()
```

Comment:
Note, I kept this call in a separate `useOnMount`. I tried combining it but felt I had to add a comment about not awaiting the result because there's no dependency between the two operations. In the end, keeping it separate felt like the best way to express the lack of dependency. (This also evolves my feelings about having multiple separate subscriptions on `authStore`.)